signal_id/noise.py

```diff
@@ -141,16 +135,27 @@ def __init__(
         else:
             warnings.warn("Noise currently requires a SpectralCube instance.")
 
-        self.spatial_footprint = np.any(self.cube.get_mask_array(),axis=0)
-
-        if isinstance(beam, Beam):
-            self.astropy_beam_flag = False
-        elif isinstance(beam, Kernel2D):
-            self.astropy_beam_flag = True
+        self.spatial_footprint = np.any(self.cube.get_mask_array(), axis=0)
+
+        if beam is None:
+            try:
+                self.beam = cube.beam
+                self.astropy_beam_flag = False
```

Comment:
shouldn't the `beam_flag` be True when there is a beam?  If not, maybe we should rename this flag to something less ambiguous?  (I haven't looked anywhere else in the code, so maybe it's obvious elsewhere)